src/test/java/net/mangolise/gamesdk/permissions/PermissionsTest.java

```diff
@@ -0,0 +1,89 @@
+package net.mangolise.gamesdk.permissions;
+
+import net.mangolise.gamesdk.permissions.backends.MapPermissionsBackend;
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.entity.Player;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.UUID;
+
+public class PermissionsTest {
+
+    @Test
+    public void testPermissions() {
+        Permissions.setBackend(new MapPermissionsBackend());
+        MinecraftServer.init();
+
+        // Connection is null for unit test
+        @SuppressWarnings("DataFlowIssue") Player player = new Player(UUID.randomUUID(), "PotatoMan", null);
+        Permissions.setPermission(player, "lovespotatoes", true);
+        Permissions.setPermission(player, "lovespotatoes.baked", true);
+        Permissions.setPermission(player, "lovespotatoes.fromcountry.*", true);
+        Permissions.setPermission(player, "lovespotatoes.fromcountry.france", false);
+        Permissions.setPermission(player, "lovespotatoes.fromcountry.fr*nce", false);
+
+        Assertions.assertTrue(Permissions.hasPermission(player, "lovespotatoes"));
+        Assertions.assertTrue(Permissions.hasPermission(player, "lovespotatoes.baked"));
+        Assertions.assertFalse(Permissions.hasPermission(player, "lovespotatoes.mashed"));
+        Assertions.assertFalse(Permissions.hasPermission(player, "lovespotatoes.fromcountry.france"));
+        Assertions.assertFalse(Permissions.hasPermission(player, "lovespotatoes.fromcountry.frHELLOWORLDnce"));
+    }
+
+    @Test
+    public void callbacks() {
+        Permissions.setBackend(new MapPermissionsBackend());
+        MinecraftServer.init();
+
+        // Connection is null for unit test
+        @SuppressWarnings("DataFlowIssue") Player player = new Player(UUID.randomUUID(), "PotatoMan", null);
+        boolean[] callbackCalled = {false};
```

Comment:
Would be much nicer to use a mocking library:

```java
import org.mockito.Mockito.*;

...

@Test
public void Test {
    Listener listener = mock(Listener.class);

    object.execute(listener);

    verify(listener).onExecuted();
}
```

But that should come in a later pr/change